cloud/amazon/ec2_group.py

```diff
@@ -163,6 +163,10 @@ def validate_rule(module, rule):
     VALID_PARAMS = ('cidr_ip',
                     'group_id', 'group_name', 'group_desc',
                     'proto', 'from_port', 'to_port')
+
+    if not isinstance(rule, dict):
```

Comment:
I'm not sure checking for Mapping/MutableMapping is valuable here - these params are always coming from inside the module, which is parsing them as a dict.